### ISSUE TITLE
Coredump during netbox:call inside on_connect callback

### ISSUE THREAD

**Issue by ochaton** [tarantool/tarantool#5611]
Tarantool version: 1.10.8-25-g3f8ed054e

OS version: MacOS

Bug description:
You need to setup any tarantool on port 3301 and run following script using another instance of tarantool

```lua
#!/usr/bin/env tarantool
local log = require 'log'
log.info(_TARANTOOL)

require'fiber'.create(function()
	local tnt = require'net.box'.connect('127.0.0.1:3301', {
		reconnect_after = 0.3,
		timeout = 1,
		wait_connected = false,
	})

	tnt:on_connect(function(self)
		log.info("Called on connected: %s:%s (%s)", self.host, self.port, self.peer_uuid)
		log.info(self:call('box.info')) -- coredump
	end)
end)

require 'console'.start() os.exit()
```

But it is safe to call `fiber.yield()` inside callback and we may execute `cnn:call` inside separate fiber newly created in callback to bypass coredump:

```lua
tnt:on_connect(function(self)
    require 'fiber'.create(function()
        log.info(self:call('box.info')) -- safe
    end)
end)
```

This bug also reproduced on 2.6.1-11-g37077a48e but instead of coredump tarantool falls into infinite loop (consumes 100% CPU and does not respond to console and signals except KILL or QUIT)

**Comment by orchaton** [comment]
Assertion message
```
Assertion failed: (last->flags & FIBER_IS_READY), function fiber_schedule_list, file /Users/v.grubov/work/tarantool/tarantool-1.10/src/fiber.c, line 629.
[1]    82331 abort      tarantool bug.lua
```

**Comment by olegrok** [comment]
Seems related to https://github.com/tarantool/tarantool/issues/5358

**Comment by locker** [comment]
Fixed in 1.10.10-1 by c7e8e6bebfb9f0331c7d2b8b3a97f4679bcb1e1a. Duplicate #6043.
